src/gsy_e/models/strategy/energy_parameters/heatpump/heat_pump.py

```diff
@@ -462,6 +498,9 @@ def _populate_state(self, time_slot: DateTime):
         )
 
     def _calculate_and_set_unmatched_demand(self, time_slot: DateTime) -> None:
-        unmatched_energy_demand = self._state.tanks.get_unmatched_demand_kWh(time_slot)
+        # pylint: disable=protected-access
+        unmatched_energy_demand = self.combined_state._charger.tanks.get_unmatched_demand_kWh(
+            time_slot
+        )
```

Comment:
I think it would be better if we remove all heat pump related variables from the tank and charger class and propose to add a method in the charger that acquires the net temperature difference and calculates the unmatched demand here. What do you think?
Nothing to do now, I can also add it in my PR